### ISSUE TITLE
Incompatible with XVim

### ISSUE THREAD

**Issue by oblitum** [wokalski/Distraction-Free-Xcode-plugin#30]
Just tried it now that Xcode 7 is supported, it indeed worked, but at the same time disabled any feature of XVim.

**Comment by wokalski** [comment]
I'm sorry to hear that. I don't use XVim. I just looked at its source and there's a lot of swizzling. Part of it is in classes instances of are not used in ZEN. This is probably the problem.

**Comment by wokalski** [comment]
I investigated it a little bit and couldn't pinpoint the issue. Only two "swizzled classes" are not used in ZEN - `IDEWorkspaceTabController` and `IDEEditorArea`. It doesn't look like there's some critical logic...

**Comment by oblitum** [comment]
Thanks for the feedback, no need to rush but if you can get it fixed sometime, I would be very glad. The "swizzled classes" thing is gibberish to me, no idea what it means xD

**Comment by oblitum** [comment]
To add more information, cursor is not visible too when entering distraction free mode, it's visible without XVim.

**Comment by oblitum** [comment]
XVim cursor is an old-style block instead of a vertical bar.

**Comment by oblitum** [comment]
Also, since you don't use XVim, you may not be aware what doesn't work. I opened the issue because none of the editing modes and Vim keybindings were working. For example, ESC seems ineffective to enter "normal mode" and walk around with hjkl keys.

**Comment by wokalski** [comment]
I installed XVim for test's sake. Let's experiment. I think many of `vim` users would like to see the distraction free mode they are used to.

**Comment by oblitum** [comment]
Indeed. From searching the web, there seems to be in fact a population of obj-c/swift developers that can't stand xcode and try to code obj-c/swift in vim. Zen-mode with XVim looks much more acceptable =)

**Comment by wokalski** [comment]
I decided not to work on this issue. It would require quite a few changes and could impact stability of the plugin. I'm leaving this open though if someone else wants to take care of it.

**Comment by wokalski** [comment]
I will also describe what I found out here.

**Comment by oblitum** [comment]
=/ OK thanks, I'm willing to work on it if I find any available time.

**Comment by wokalski** [comment]
More info:

XVim compatibility:
- Editor - XVim editor heavily relies on `IDEEditorArea`. It's a view controller _above_ `IDEEditorContext` in stack. It means that it is not used in ZEN since ZEN relies only on `IDEEditorContext`.
- Layout - XVim let's you use vi bindings. It also includes editor layout management. It's performed using method swizzling/some magic. They use `IDEWorkspaceTabController` to perform those modifications. 

`IDEEditorContext` also relies heavily on the tab controller to resolve some dependencies. It is currently mocked using `ZENIDEEditorContextDependencyManager`. This class also contains some mocking of `IDEWorkspaceWindowController`. This is a mess.

**Comment by wokalski** [comment]
Oh, and to whoever works on this: just ask if you need more info!

**Comment by jordwalke** [comment]
I'd love to see this fixed since Distraction Free Mode and Vim kind of go hand in hand!

**Comment by oblitum** [comment]
I had to choose between the two for now and had to stay with XVim :-(

**Comment by yam-liu** [comment]
so sad.
